test/parallel/test-domain-timers-uncaught-exception.js

```diff
@@ -0,0 +1,14 @@
+const domain = require('domain').create();
+var first = false;
+
```

Comment:
Can't comment much on the other change, but the test needs a `'use strict'`, a `require('../common')`, and a little comment explaining what this tests would be nice (though not actually required).

If you haven't seen the [guide to writing tests](https://github.com/nodejs/node/blob/master/doc/guides/writing-tests.md) then I recommend reading through it.